### PR TITLE
Updated the release branch docs

### PR DESCRIPTION
**Pull request by vincentpierre**
### Proposed change(s)

Updating the Readme, Changelog and Migrating on the release branch

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 <img src="docs/images/image-banner.png" align="middle" width="3000"/>
 
 # Unity ML-Agents Toolkit (Beta)
-[![docs badge](https://img.shields.io/badge/docs-reference-blue.svg)](https://github.com/Unity-Technologies/ml-agents/tree/latest_release/docs/)
+[![docs badge](https://img.shields.io/badge/docs-reference-blue.svg)](https://github.com/Unity-Technologies/ml-agents/tree/0.15.0/docs/)
```
Comment (surfnerd) [inline README.md]: why this change? I thought we always wanted it to point to `latest_release`.

File: README.md [hunk 2]
```diff
@@ -61,8 +61,8 @@ details of the changes between versions.
 
 | **Version** | **Release Date** | **Source** | **Documentation** | **Download** |
 |:-------:|:------:|:-------------:|:-------:|:------------:|
-| **master** (unstable) | -- | [source](https://github.com/Unity-Technologies/ml-agents/tree/master) |  [docs](https://github.com/Unity-Technologies/ml-agents/tree/master/docs/Readme.md) | [download](https://github.com/Unity-Technologies/ml-agents/archive/master.zip) |
```
Comment (surfnerd) [inline README.md]: Also, why the removal of master (unstable)?

File: README.md (hunk 1 above)
Comment (vincentpierre) [reply README.md]: when a user goes on the branch with the tag 0.15, the docs should point towards 0.15.
I think we only want to point to latest release on master but on a release branch, we want to point to the release branch.

**Comment by surfnerd** [comment]
not sure why all of the `latest_release` links were removed.  Please change those back to `latest_release` for the 0.15.0 links.  And add back the `master (unstable)` doc links.

File: README.md (hunk 2 above)
Comment (vincentpierre) [reply README.md]: This is for the release branch. II do not think the release branch should point to the `moving` master/latest_release

Comment (surfnerd) [reply README.md]: I think having the "master (unstable)" link should always be there.  

Will you change the 0.15.0 -> `latest_release` link before you actually create the release?

File: README.md (hunk 1 above)
Comment (chriselion) [reply README.md]: This will cause the markdown link check to fail (although only the nightly job checks remote links) since the tag doesn't exist.

File: README.md (hunk 2 above)
Comment (chriselion) [reply README.md]: The intent when I added this was to only link to previous releases, not master or `latest_release`
